blob/azureblob/azureblob.go

```diff
@@ -208,12 +209,28 @@ func openerFromEnv(accountName AccountName, accountKey AccountKey, sasToken SAST
 
 // OpenBucketURL opens a blob.Bucket based on u.
 func (o *URLOpener) OpenBucketURL(ctx context.Context, u *url.URL) (*blob.Bucket, error) {
-	for k := range u.Query() {
-		return nil, fmt.Errorf("open bucket %v: invalid query parameter %q", u, k)
+	err := o.setOptionsFromURLParams(u.Query())
```

Comment:
This has a side effect of changing the Option for all future URLs opened by the URLOpener. Please clone the o.Options first, similar to:

https://github.com/google/go-cloud/blob/master/blob/gcsblob/gcsblob.go#L232

You can clone it, and then pass it to setOptionsFromURLParams, which will no longer need to be a function of the struct.